### PR TITLE
Migrate the app to support the front-end and back-end

### PR DESCRIPTION
**Pull request by Munikumar09**
This pull request migrates the app to support the separation of front-end and back-end code. This separation will improve maintainability and allow for easier updates and enhancements in the future.

## Release Notes

- **New Features**
	- Introduced scripts for formatting and linting Python files, enhancing code quality checks.
	- Added handling for JSON file creation based on stock data intervals in the data downloader utilities.

- **Bug Fixes**
	- Improved error handling in various functions to ensure robust responses to invalid inputs and exceptions.

- **Documentation**
	- Updated README and comments for clarity on usage and configuration.

- **Tests**
	- Enhanced test coverage for various modules, including additional test cases for error handling and data processing functions.

- **Chores**
	- Reorganized import statements across multiple files for improved readability and consistency.

### CODE REVIEW

File: backend/scripts/setup [hunk 1]
```diff
@@ -24,21 +25,19 @@ def setup_env_vars():
...
 
     # Create deactivation script
     with open(deactivate_dir / "project_env_vars.sh", "w") as f:
         f.write("#!/bin/bash\n")
         f.write("unset ROOT_PATH\n")
 
-    with open(project_root.parent / ".env", "w") as f:
-        f.write("ROOT_PATH=" + str(project_root) + "\n")
+    with open(BACKEND_ROOT_PATH.parent / ".env", "w") as f:
+        f.write("ROOT_PATH=" + str(BACKEND_ROOT_PATH) + "\n")
```
[the remark this answers is not in the document]
Comment (Munikumar09) [reply backend/scripts/setup:40]: Changes were done

File: backend/scripts/setup [hunk 2]
```diff
@@ -49,6 +48,7 @@
     """Installs dependencies using Poetry."""
     try:
         logger.info("Installing dependencies with Poetry...")
+        os.chdir(BACKEND_ROOT_PATH)
         subprocess.run(["poetry", "install"], check=True)
```
[the remark this answers is not in the document]
Comment (Munikumar09) [reply backend/scripts/setup:52]: Done

File: backend/scripts/lint.sh [hunk 3]
```diff
@@ -1,6 +1,12 @@
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+cd $SCRIPT_DIR
 cd ..
+echo "Current script parent path: $(pwd)"
+
+
 echo "Running mypy check..."
 mypy $(git ls-files "*.py")
```
[the remark this answers is not in the document]
Comment (Munikumar09) [reply backend/scripts/lint.sh]: Done

File: .github/workflows/fastapi-test.yaml [hunk 4]
```diff
@@ -36,21 +36,17 @@ jobs:
...
-        run: isort --diff --check --profile black $(git ls-files '*.py')
-      - name: mypy check
-        run: mypy $(git ls-files '*.py')
-      - name: pylint check
-        run: pylint $(git ls-files '*.py')
-
+      - name: Format check
+        run: ./scripts/format.sh --check
+      - name: Lint check
+        run: ./scripts/lint.sh
```
[the remark this answers is not in the document]
Comment (munikumar-mm-17774) [reply .github/workflows/fastapi-test.yaml:42]:
```suggestion
        run: ./scripts/format.sh --check
      - name: Lint check
        run: ./scripts/lint.sh
```

File: backend/scripts/lint.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    git ls-files "*.py" | xargs mypy
+}
+
+run_pylint_check() {
+    echo "Running pylint check..."
+    git ls-files "*.py" | xargs pylint
+}
+
+run_mypy_check
+run_pylint_check
```
[the remark this answers is not in the document]
Comment (munikumar-mm-17774) [reply backend/scripts/lint.sh]: Done

File: backend/scripts/lint.sh [hunk 6]
```diff
@@ -1,6 +1,12 @@
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+cd $SCRIPT_DIR
```
[the remark this answers is not in the document]
Comment (munikumar-mm-17774) [reply backend/scripts/lint.sh]: Done

File: backend/scripts/format.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+cd $SCRIPT_DIR
+cd ..
```
[the remark this answers is not in the document]
Comment (Munikumar09) [reply backend/scripts/format.sh]: Done

File: .github/workflows/fastapi-test.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Munikumar09) [reply .github/workflows/fastapi-test.yaml:42]: Done

**Pull request merged**